### ISSUE TITLE
`console.warn` variable reference doesn't get disabled with `eslint-disable-line no-console`

### ISSUE THREAD

**Issue by danfuzz** [eslint/eslint#9327]
**Tell us about your environment**

* **ESLint Version:** 4.7.0
* **Node Version:** 8.4.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": "eslint:recommended",
  "env": {
    "es6": true,
    "node": true
  },
  "parserOptions": {
    "sourceType": "module",
    "ecmaVersion": "2017"
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
export default class ConsoleFun {
  foo(level) {
    let logMethod;
    switch (level) {
      case 'error': { logMethod = console.error; break; } // eslint-disable-line no-console
      case 'warn':  { logMethod = console.warn;  break; } // eslint-disable-line no-console
      default:      { logMethod = console.log;   break; } // eslint-disable-line no-console
    }

    return logMethod;
  }

  bar(level) {
    switch (level) {
      case 'error': return console.error; // eslint-disable-line no-console
      case 'warn':  return console.warn;  // eslint-disable-line no-console
      default:      return console.log;   // eslint-disable-line no-console
    }
  }
}
```

**What did you expect to happen?**
No linter error.

**What actually happened? Please include the actual, raw output from ESLint.**
```
$ eslint ConsoleFun.js 

/Users/dbornstein/testing/lint-fun/ConsoleFun.js
  6:35  error  Unexpected console statement  no-console

✖ 1 problem (1 error, 0 warnings)
```

**Note:** If you reverse the order of the two methods so that `bar` is first, then the `console.warn` line in `bar` is the one that gets reported.

**Comment by not-an-aardvark** [comment]
Closing as a duplicate of https://github.com/eslint/eslint/issues/9318.
